semgrep/semgrep/output.py

```diff
@@ -243,9 +245,24 @@ def handle_semgrep_errors(self, errors: List[SemgrepError]) -> None:
     def handle_semgrep_timeout_errors(self, errors: Dict[Path, List[str]]) -> None:
         self.has_output = True
         separator = ", "
+        print_threshold_hint = False
         for path in errors.keys():
+            num_errs = len(errors[path])
+            errors[path].sort()
+            error_msg = f"Warning: {num_errs} timeout error(s) in {path} when running the following rules: [{separator.join(errors[path])}]"
+            if num_errs == self.settings.timeout_threshold:
+                error_msg += f"\nSemgrep stopped running rules on {path} after {num_errs} timeout error(s). See `--timeout-threshold` for more info."
+            print_threshold_hint = print_threshold_hint or (
+                num_errs > 5 and not self.settings.timeout_threshold
+            )
+            logger.error(with_color(colorama.Fore.RED, error_msg))
```

Comment:
Do we typically use `with_color` with our loggers? I see it's only currently used in `semgrep/error.py` and not in any `logger` calls. I'll defer to @brendongo here.

Reply:
I used it here because the messages were red before, so when aggregating them and creating a new error message I added `with-color` to make them look like the other not aggregated error messages.

Reply:
I think there are other errors that have color added as part of their __str__ function that get written to log so this is fine.